pkgs/by-name/fl/flood-for-transmission/package.nix

```diff
@@ -1,39 +1,51 @@
 # To use this package, use: `services.transmission.webHome = pkgs.flood-for-transmission;`
+# The configuration [1] can be modified by overriding floodSettings:
+# pkgs.flood-for-transmission.override {
+#   floodSettings.SWITCH_COLORS = true;
+# };
+# [1]: https://github.com/johman10/flood-for-transmission?tab=readme-ov-file#beta-customization
 {
   lib,
   buildNpmPackage,
   fetchFromGitHub,
+  floodSettings ? null,
+  nix-update-script,
 }:
 
-buildNpmPackage rec {
+buildNpmPackage (finalAttrs: {
   pname = "flood-for-transmission";
   version = "2025-07-19T10-51-22";
 
   src = fetchFromGitHub {
     owner = "johman10";
     repo = "flood-for-transmission";
-    tag = version;
+    tag = finalAttrs.version;
     hash = "sha256-2oHEVvZZcxH9RBKreaiwFKp7Iu5ijYdpXdgVknCxwKw=";
   };
 
   npmDepsHash = "sha256-IUdsUGsm6yAbXqf4UGkz1VPa366TnWsTakkbywbLeTU=";
 
-  strictDeps = true;
-
   installPhase = ''
     runHook preInstall
 
     cp -r public $out
+    rm $out/config.json.defaults
 
     runHook postInstall
   '';
 
-  meta = with lib; {
+  postInstall = lib.optionalString (floodSettings != null) ''
+    echo '${builtins.toJSON floodSettings}' > $out/config.json
```

Comment:
Should it be merged with `config.json.defaults`?

Reply:
IIUC flood already merges the settings in config.json with the defaults: https://github.com/johman10/flood-for-transmission/blob/e8f0f066b0a39680b9511a6f1db58c15e6a3a231/src/helpers/configHelper.js#L56